pandas/tests/indexes/test_setops.py

```diff
@@ -463,3 +466,33 @@ def test_setop_with_categorical(index, sort, method):
     result = getattr(index, method)(other[:5], sort=sort)
     expected = getattr(index, method)(index[:5], sort=sort)
     tm.assert_index_equal(result, expected)
+
+
+@pytest.mark.parametrize("index_maker", tm.index_subclass_makers_generator())
+@pytest.mark.parametrize("reverse", [True, False])
+def test_valid_intersection_w_dupes(index_maker, reverse):
+    # TODO: it would be good to ensure these are unique (categoricals aren't)
+    full = index_maker(k=4)
+    series = [pd.Series(1, index=full[[0, 1, 1, 2]]), pd.Series(0, index=full[[0, 2]])]
+    if reverse:
+        series = reversed(series)
+
+    result = get_objs_combined_axis(series, intersect=True)
+
+    tm.assert_index_equal(full[[0, 2]], result, check_order=False)
```

Comment:
always use 
result = 
expected = 
tm.assert_index(result, expected)

why is check_order=False?

Reply:
`check_order` is false because the `MultiIndex` comes back in a different order.

More conceptually, I don't think this function promises any ordering unless you pass sort.